bazel/integration/test_runner/package_json.ts

```diff
@@ -82,6 +106,24 @@ function updateMappingForRecord(record: DependencyRecord, mappings: PackageMappi
       continue;
     }
 
-    record[pkgName] = mappedAbsolutePath;
+    record[entryKey] = mappedAbsolutePath;
+  }
+}
+
+/**
+ * Gets the package name from a dependency record entry.
+ *
+ * @param entryKey Key of the dependency record entry.
+ * @param nameMatchRegex Optional regular expression that can be specified to match the package
+ *   name in a dependency entry. This is useful for e.g. Yarn resolutions using patterns.
```

Comment:
Might be worth mentioning that the value matched by the first capturing group of the regex is used as the package name.

Reply:
agreed 👍